src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -584,17 +584,13 @@ ucs_status_t uct_rc_verbs_ep_get_address(uct_ep_h tl_ep, uct_ep_addr_t *addr)
     uct_rc_verbs_ep_t *ep                 = ucs_derived_of(tl_ep, uct_rc_verbs_ep_t);
     uct_ib_md_t *md                       = uct_ib_iface_md(&iface->super.super);
     uct_rc_verbs_ep_flush_addr_t *rc_addr = (uct_rc_verbs_ep_flush_addr_t*)addr;
-    uint8_t mr_id;
 
     rc_addr->super.flags = 0;
     uct_ib_pack_uint24(rc_addr->super.qp_num, ep->qp->qp_num);
 
-    if (md->ops->get_atomic_mr_id(md, &mr_id) == UCS_OK) {
-        ucs_assertv(uct_ib_md_is_flush_rkey_valid(md->flush_rkey),
-                    "invalid flush_rkey %x for device %s", md->flush_rkey,
-                    uct_ib_device_name(&md->dev));
+    md->ops->get_atomic_mr_id(md, &rc_addr->atomic_mr_id);
```

Comment:
looks error prone trying to fill the address without knowing whether atomic mr is supported

Reply:
@brminich If it's not supported, it still needs to assign value(e.g. 0)
https://github.com/openucx/ucx/blob/v1.15.0-rc1/src/uct/ib/mlx5/ib_mlx5.c#L951

Reply:
if it is not supported you may use smaller address and will try to access invalid memory.

Reply:
Under this case, it won't set ```UCT_RC_VERBS_ADDR_HAS_ATOMIC_MR``` flag, right?

Reply:
now ep address length (and thus its format) is defined as below
```
iface_attr->ep_addr_len = uct_rc_iface_flush_rkey_enabled(&iface->super) ?
 sizeof(uct_rc_verbs_ep_flush_addr_t) : sizeof(uct_rc_verbs_ep_addr_t);
```
which means that we are not supposed to access `rc_addr->atomic_mr_id` if flush rkey is not enabled (e.g. outside of the if below).

Reply:
@brminich Good catch.
@yosefe @brminich Let me fix this point.

Reply:
@brminich It's fixed. I also checked DC, there's no similar problem.